src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -172,6 +206,70 @@ public static Text of(Object... objects) {
     }
 
     /**
+     * Creates a new Text instance with all {@link Placeholder}s replaced. All
+     * placeholders without a non-null replacement are ignored. All replacements
+     * will be wrapped in a {@link Text} using {@link Texts#of(Object...)} the
+     * color and the style from the placeholder are transfered to that method as
+     * well.
+     *
+     * @param template The template text in which all {@link Placeholder}s
+     *        should be replaced
+     * @param replacements The values available to replace the placeholders
+     * @return The text with all possible placeholders replaced
+     */
+    public static Text format(Text template, Map<String, ?> replacements) {
+        checkNotNull(template, "template");
+        checkNotNull(replacements, "values");
+        if (replacements.isEmpty()) {
+            return template;
+        }
+        return formatNoChecks(template, replacements);
+    }
+
+    /**
+     * Creates a new Text instance with all {@link Placeholder}s replaced. All
+     * placeholders without a non-null replacement are ignored. All replacements
+     * will be wrapped in a {@link Text} using {@link Texts#of(Object...)} the
+     * color and the style from the placeholder are transfered to that method as
+     * well.
+     *
+     * @param template The template text in which all {@link Placeholder}s
+     *        should be replaced
+     * @param replacements The values available to replace the placeholders. May
+     *        contain null values to skip the placeholder
+     * @return The text with all possible placeholders replaced
+     */
+    public static Text format(Text template, Object... replacements) {
+        checkNotNull(template, "template");
+        checkNotNull(replacements, "values");
+        Map<String, Object> replacementsMap = new HashMap<String, Object>();
+        int index = 0;
+        for (Object replacement : replacements) {
+            replacementsMap.put(Integer.toString(index++), replacement);
+        }
+        return formatNoChecks(template, replacementsMap);
+    }
+
+    private static Text formatNoChecks(Text template, Map<String, ?> replacements) {
+        // Is this a placeholder that should be replaced?
+        if (template instanceof Placeholder) {
+            Object replacement = replacements.get(((Placeholder) template).getKey());
+            // Only replace
+            if (replacement != null) {
+                // Copy color and style from placeholder
+                return Texts.of(template.getColor(), template.getStyle(), replacement);
```

Comment:
@zml2008
> also click, hover, etc.

Click and Hover are not supported by Texts.of, 
but i will add that too. I did not add that before because IMO the textactions are always provided by the replacements itself.
A clickaction like OpenLink does not make sense with a variable text. And a hover text usually shows a hint/description for the object it describes.

> And if the replacement is a Text, you might want to have it just return the replacement itself

No, because in that case the color and style... are lost. This would be inconsistent with the rest of the method.